pkgs/applications/science/electronics/kicad/default.nix

```diff
@@ -187,27 +190,34 @@ stdenv.mkDerivation rec {
 
   # We are emulating wrapGAppsHook, along with other variables to the
   # wrapper
-  makeWrapperArgs = with passthru.libraries; [
-    "--prefix XDG_DATA_DIRS : ${base}/share"
-    "--prefix XDG_DATA_DIRS : ${hicolor-icon-theme}/share"
-    "--prefix XDG_DATA_DIRS : ${gnome3.defaultIconTheme}/share"
-    "--prefix XDG_DATA_DIRS : ${wxGTK.gtk}/share/gsettings-schemas/${wxGTK.gtk.name}"
-    "--prefix XDG_DATA_DIRS : ${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}"
-    # wrapGAppsHook did these two as well, no idea if it matters...
-    "--prefix XDG_DATA_DIRS : ${cups}/share"
-    "--prefix GIO_EXTRA_MODULES : ${gnome3.dconf}/lib/gio/modules"
+  makeWrapperArgs =
+    let
+      # Use user-provided symbol, footprint and 3D package locations if set.
+      symbols = if symbolsPath != null then symbolsPath else symbols;
```

Comment:
You have a whole `let in` section up there were you can do what you want.

Reply:
Actually my commit shouldn't work because `symbols` etc aren't defined here. I need `passthru.libraries.symbols`. But won't this prevent doing this in the upper let? That's why I placed this here.

Reply:
That's why `with` is an anti pattern. Get rid of it, and put in `passthru.libraries.symbols` the real value according to the override.

Reply:
Ok I've made a change. I'm not confident I've achieved what you're asking for. Would you prefer that I change `libraries.nix` so that symbols can be defined directly in `passthru.libraries`?

Reply:
I think changing `libraries.nix` is the only way to be able to define `passthru.libraries.symbols=...` to incorporate the override, but maybe I'm missing something?

Reply:
> I think changing `libraries.nix` is the only way to be able to define `passthru.libraries.symbols=...`

It's not. _If_ this was what I'd have recommend to do, you'd have had to use something like:

```nix
passthru.libraries = (callPackages ./libraries.nix { }) // { symbols = ...; };